suite-common/connect-popup/src/methodHooks/ethereumSignTransaction.ts

```diff
@@ -0,0 +1,111 @@
+import { Bip43Path, getNetworkByEvmChainId } from '@suite-common/wallet-config';
+import {
+    accountsActions,
+    selectAccounts,
+    selectSelectedDevice,
+    sendFormActions,
+} from '@suite-common/wallet-core';
+import TrezorConnect from '@trezor/connect';
+import type { EthereumSignTransaction } from '@trezor/connect';
+import { getSerializedPath, validatePath } from '@trezor/connect/src/utils/pathUtils';
+
+import { connectPopupActions } from '../connectPopupActions';
+
+import { PreCallHookParams } from './index';
+
+const preCallHook = async <M extends keyof typeof TrezorConnect>({
+    method,
+    payload,
+    getState,
+    dispatch,
+    txSigningPrecomposed,
+}: PreCallHookParams<M>) => {
+    try {
+        if (method === 'ethereumSignTransaction' && txSigningPrecomposed) {
+            const typedPayload = payload as any as EthereumSignTransaction;
+            const device = selectSelectedDevice(getState());
```

Comment:
I wonder how connect call device param and selecting currently selected device work together? is connect call device param solved at some earlier point and we can presume that selectSelectedDevice returns correct device?

Reply:
True, we should probably use device from the call. We just need it for the placeholder account